app/controllers/matchups_controller.rb

```diff
@@ -4,10 +4,56 @@ class MatchupsController < ApplicationController
   def set_score
   end
 
+  def create
+    @matchup = Matchup.new(matchup_params)
+    @tournament = @matchup.player1.tournament
+
+    if @matchup.save!
+      redirect_to edit_tournament_path(@tournament)
+    else
+      render "tournaments/edit", status: :unprocessable_entity
+    end
+
+    authorize @matchup, policy_class: MatchupPolicy
+  end
+
   def update
     if @matchup.update(matchup_params)
+      # Update the wins and losses of each player
+      # based on the submitted scores
+      p1 = @matchup.player1
+      p2 = @matchup.player2
+      if @matchup.player1_score > @matchup.player2_score
+        p1.win_count += 1
+        p2.loss_count += 1
+      elsif @matchup.player1_score < @matchup.player2_score
+        p1.loss_count += 1
+        p2.win_count += 1
+      else
+        p1.win_count += 0.5
+        p1.loss_count += 0.5
+        p2.win_count += 0.5
+        p2.loss_count += 0.5
+      end
+
+      p1.save!
+      p2.save!
+
+      # Check if all the scores have been submitted
+      # for the round and if so, generate matchups
+      # for this division two rounds later
+      this_tournament = Tournament.find(p1.tournament_id)
+      matchups_without_scores = this_tournament.matchups.where(round_number: @matchup.round_number).where(player1_score: nil).to_a
```

Comment:
Does this work as `this_tournament.matchups.where(round_number: @matchup.round_number, player1_score: nil).to_a` ?